Kernel/Arch/x86_64/PCI/Initializer.cpp

```diff
@@ -52,35 +112,21 @@ UNMAP_AFTER_INIT void initialize()
     }
     if (g_pci_access_is_disabled_from_commandline || g_pci_access_io_probe_failed)
         return;
+
+    PCI::Access::initialize();
     switch (detect_optimal_access_type()) {
     case PCIAccessLevel::MemoryAddressing: {
         VERIFY(possible_mcfg.has_value());
-        auto success = Access::initialize_for_multiple_pci_domains(possible_mcfg.value());
-        VERIFY(success);
+        MUST(find_and_register_pci_host_bridges_from_acpi_mcfg_table(possible_mcfg.value()));
         break;
     }
     case PCIAccessLevel::IOAddressing: {
-        auto success = Access::initialize_for_one_pci_domain();
-        VERIFY(success);
+        MUST(initialize_for_one_pci_domain());
         break;
     }
     default:
         VERIFY_NOT_REACHED();
     }
-
-    // IRQ from pin-based interrupt should be set as reserved as soon as possible so that the PCI device
-    // that chooses to use MSI(x) based interrupt can avoid sharing the IRQ with other devices.
-    MUST(PCI::enumerate([&](DeviceIdentifier const& device_identifier) {
-        // A simple sanity check to avoid getting a panic in get_interrupt_handler() before setting the IRQ as reserved.
-        if (auto irq = device_identifier.interrupt_line().value(); irq < GENERIC_INTERRUPT_HANDLERS_COUNT) {
-            auto& handler = get_interrupt_handler(irq);
-            handler.set_reserved();
-        }
-    }));
-
-    MUST(PCI::enumerate([&](DeviceIdentifier const& device_identifier) {
```

Comment:
You probably should also remove this printing loop from riscv64/PCI/Initializer.cpp as well.